ccmpred/scripts/run_ccmgen.py

```diff
@@ -212,9 +212,22 @@ def main():
                 neff = ccm.neff_entropy
             else:
                 neff = opt.neff
-
-            msa_sampled, neff = ccmpred.sampling.sample_to_neff_increasingly(
-                tree, nseq, neff, ncol, x, opt.seq0_mrf)
+            
+            if opt.seq0_file:
+                
+                seq0 = ccmpred.io.alignment.read_msa(opt.seq0_file, opt.aln_format)
+                print("Ancestor sequence:\n{0}".format("".join([ccmpred.io.alignment.AMINO_ACIDS[c] for c in seq0[0]])))
+                if seq0[0].shape[0] != ncol:
+                    print("Length of ancestor sequence must match dimension of MRF model!")
+                    exit(0)
+                    
+                msa_sampled, neff = ccmpred.sampling.sample_to_neff_increasingly(
+                tree, nseq, neff, ncol, x, opt.seq0_mrf, seq0=seq0)
+                
+            else:
+                
+                msa_sampled, neff = ccmpred.sampling.sample_to_neff_increasingly(
```

Comment:
The two function calls to `ccmpred.sampling.sample_to_neff_increasingly` should be equal, if you initialise `seq0 = None` before the `if opt.seq0_file` block, right? so you should be able to save the duplicated call.

Reply:
Done